workflow/controller/operator.go

```diff
@@ -1338,6 +1338,16 @@ func (woc *wfOperationCtx) executeTemplate(nodeName string, orgTmpl wfv1.Templat
 		if err != nil {
 			return woc.markNodeError(retryNodeName, err), err
 		}
+		if lastChildNode != nil && lastChildNode.Pending() && processedTmpl.GetType() == wfv1.TemplateTypeContainer {
+			_, err := woc.createWorkflowPod(lastChildNode.Name, *processedTmpl.Container, processedTmpl, false)
```

Comment:
this is a significant piece of code - can you please add some tests?

Reply:
@alexec Yeah, I've been thinking to add e2e tests, and it requires to set a namespace with quota, let me try.